### DISCUSSION TITLE
Is it possible to generate the proof verifier in bitcoin script

### DISCUSSION THREAD

**Discussion by zhengjunhe** [Consensys/gnark#962] [Ideas]
Hi

Is it possible to generate the proof verifier in bitcoin script just as the vefifier in solidity code for ethereum.
If it does,gnark can be used in bitcoin network.

Thanks .

**Comment by zhengjunhe** [comment]
Described as in the link below:
https://xiaohuiliu.medium.com/create-your-first-zero-knowledge-proof-program-on-bitcoin-ec159cc501f4

ZoKrates is a frontend tool to create zkp proof and it can export verifier in solidity.Script extends the ZoKrates that make it can export verifier in btc script.As I have developed some circuits based on gnark,if gnark could also export verifier in btc script,I think it will be loved by zkp application developers.

Thanks.

**Comment by gbotrel** [comment]
Hey -- thanks for the message. Keeping that in mind, but unfortunately we don't have the bandwidth to tackle that in the short term.
